config/www/light-map-panel.js

```diff
@@ -48,13 +77,191 @@ class LightMapPanel extends LitElement {
+    const roomsGroup = svgDoc.querySelector("#Rooms");
+
+    if (!roomsGroup) return null;
+
+    const transform = roomsGroup.getAttribute("transform");
+    let translateX = 0;
+    let translateY = 0;
+    if (transform) {
+      const match = /translate\((-?\d+\.?\d*),\s*(-?\d+\.?\d*)\)/.exec(transform);
+      if (match) {
+        translateX = parseFloat(match[1]);
+        translateY = parseFloat(match[2]);
+      }
+    }
+
+    const adjustedX = x - translateX;
+    const adjustedY = y - translateY;
+
+    const roomElements = roomsGroup.querySelectorAll("g[id]");
+    for (const room of roomElements) {
+      const rect = room.querySelector("rect");
+      if (!rect) continue;
+
+      const roomX = parseFloat(rect.getAttribute("x"));
+      const roomY = parseFloat(rect.getAttribute("y"));
+      const roomWidth = parseFloat(rect.getAttribute("width"));
+      const roomHeight = parseFloat(rect.getAttribute("height"));
+
+      if (
+        adjustedX >= roomX &&
+        adjustedY >= roomY &&
+        adjustedX <= roomX + roomWidth &&
+        adjustedY <= roomY + roomHeight
+      ) {
+        return room.id;
+      }
+    }
+
+    return null;
+  }
+
+  handleSVGClick(event) {
+    if (!this.selectedSensor) {
+      alert("Please select a sensor first.");
+      return;
+    }
+
+    const svgContainer = this.shadowRoot.querySelector(".svg-container svg");
+    if (!svgContainer) {
+      return;
+    }
+
+    //Get the click position in the SVG coordinates
+    const point = svgContainer.createSVGPoint();
+    point.x = event.clientX;
+    point.y = event.clientY;
+    const svgPoint = point.matrixTransform(svgContainer.getScreenCTM().inverse());
+
+    const { x, y } = svgPoint;
+
+    //Ensure the placement is inside a room
+    const roomId = this.getRoomFromCoordinates(x, y);
+    if (!roomId) {
+      alert("Sensor placement is outside of any room.");
+      return;
+    }
+
+    //Check if the sensor is already placed
+    if (this.isSensorAlreadyPlaced(this.selectedSensor.id)) {
+      alert("This sensor has already been placed.");
+      return;
+    }
+
+    //Generate a random color for the sensor
+    const randomColor = `#${Math.floor(Math.random() * 16777215).toString(16)}`;
+
+    //Add sensor to the placedSensors list
+    if (!this.placedSensors[roomId]) {
+      this.placedSensors[roomId] = [];
+    }
+    this.placedSensors[roomId].push({
+      ...this.selectedSensor,
+      x,
+      y,
+      color: randomColor,
+    });
+
+    //Parse the SVG and add the circle and wavy lines
+    const parser = new DOMParser();
+    const svgDoc = parser.parseFromString(this.uploadedSVG, "image/svg+xml");
+
+    //Create a group to hold the sensor elements (circle and wavy lines)
+    const groupElement = svgDoc.createElementNS("http://www.w3.org/2000/svg", "g");
+    groupElement.setAttribute("transform", `translate(${x}, ${y})`);
+    groupElement.setAttribute("data-sensor-id", this.selectedSensor.id);
```

Comment:
Is it possible to rename this to: "id"?

The svg_accessor utilizes the "id" field to locate elements in the svg.

Reply:
Fixed in commit